mesh_segmenter/src/convex_hull.cpp

```diff
@@ -0,0 +1,101 @@
+#include <ros/ros.h>
+#include "noether_msgs/GenerateConvexHull.h"
+#include <fstream>
+#include <string>
+#include <pcl/PolygonMesh.h>
+#include <pcl/point_types.h>
+#include <pcl/conversions.h>
+#include <pcl/io/ply_io.h>
+#include <pcl/io/vtk_lib_io.h>
+#include <pcl/surface/convex_hull.h>
+#include <pcl/common/centroid.h>
+#include <Eigen/Dense>
+
+
+using namespace std;
+
+
+
+bool generate_ch(noether_msgs::GenerateConvexHull::Request& req,
+	                 noether_msgs::GenerateConvexHull::Response& res)
+{
+
+  string inMeshFileName = req.file_in; //expects a path to file
+  string modifier_ = "_chullt.ply";
+  string outMeshFileName = inMeshFileName.substr(0, inMeshFileName.size()-4);
+  outMeshFileName.append(modifier_);
+  
+  pcl::PointCloud<pcl::PointXYZ>::Ptr inMesh (new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PointCloud<pcl::PointXYZ>::Ptr outMesh (new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PolygonMesh::Ptr outMeshPoly (new pcl::PolygonMesh);
+
+  
+  //if its a .ply file
+  pcl::PLYReader Reader;
+  Reader.read(inMeshFileName, *inMesh); //populate inMesh
```

Comment:
Some file existence checking would be nice here,  you can use boost/filesystem.hpp